src/tracer.js

```diff
@@ -253,6 +270,32 @@ export default class Tracer {
     }
 
     /**
+     * Reapplies sampling if the parent span isn't sampled
+     *
+     * @param parent the parent span context
+     * @param ctx the span context
+     * @param operationName the operation name for the child
+     * @param tags tags to be applied by the sampler
+     */
+    handleUpsampling(parent: SpanContext, ctx: SpanContext, operationName: string, tags: any) {
+        if (!this._upsampling.enabled) {
+            return
+        }
+
+        // Reuse upsampling decision, if one exists in parent
+        if (parent.upsamplingDecision) {
+            ctx.samplingDecision = parent.upsamplingDecision.valueOf();
+            ctx.upsamplingDecision = parent.upsamplingDecision;
+            return
+        }
+
+        // Store upsampling decision in parent
+        parent.upsamplingDecision = new Boolean(this._sampler.isSampled(operationName, tags));
+        ctx.samplingDecision = parent.upsamplingDecision.valueOf();
+        ctx.upsamplingDecision = parent.upsamplingDecision;
+    }
```

Comment:
this doesn't look right.
  * I thought upsampling and finalized will be the same now. When do they need to be different?
  * the propagation of upsampling into the parent in L293 looks just wrong, why do we want that?
  * I am not seeing the conditional of only allowing upsampling on first in process span.